src/ui/camera.js

```diff
@@ -546,7 +575,9 @@ class Camera extends Evented {
                 tr.pitch = interpolate(startPitch, pitch, k);
             }
 
-            tr.setLocationAtPoint(toLngLat, fromPoint.add(toPoint.sub(fromPoint)._mult(k)));
+            var lng = interpolate(startCenter.lng, toLocation.lng, k2);
+            var lat = interpolate(startCenter.lat, toLocation.lat, k2);
+            tr.center = LngLat.convert([lng, lat]);
```

Comment:
Is it possible to write test for this new functionality? Regression tests for https://github.com/mapbox/mapbox-gl-js/issues/3112 & https://github.com/mapbox/mapbox-gl-js/issues/4178?